tftypes/value_json.go

```diff
@@ -16,7 +16,28 @@ import (
 // terraform-plugin-go.  Third parties should not use it, and its behavior is
 // not covered under the API compatibility guarantees. Don't use this.
 func ValueFromJSON(data []byte, typ Type) (Value, error) {
-	return jsonUnmarshal(data, typ, NewAttributePath())
+	return jsonUnmarshal(data, typ, NewAttributePath(), JSONOpts{})
+}
+
+// UnmarshalOpts contains options that can be used to modify the behaviour when
+// unmarshalling. Currently, this only contains a struct for opts for JSON but
+// could have a field for Flatmap in the future.
+type UnmarshalOpts struct {
```

Comment:
😭 Worst nit ever... could we move this to both the tfprotov5 and tfprotov6 packages so its directly tied with the `Unmarshal` function its providing options for? I'm _very_ doubtful we would have differing implementations between the two protocol versions, but I'd hate to be wrong and its hopefully not an expensive choice to keep them in sync as necessary.

Reply:
Makes sense. Have moved.